middleware/validateID.js

```diff
@@ -0,0 +1,19 @@
+require("express-async-errors");
+const mongoose = require("mongoose");
+
+module.exports = async(req, res, next) => {
+  const arrayOfId = [];
```

Comment:
Can you clarify why an array is used in this code? For validating a single ObjectId, would it not be sufficient to use `const isValid = mongoose.Types.ObjectId.isValid(providedId);`?